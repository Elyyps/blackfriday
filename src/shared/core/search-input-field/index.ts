export * from "./search-input-field.component";
