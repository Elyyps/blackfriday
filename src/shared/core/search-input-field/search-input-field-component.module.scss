.search-input-field {
  width: 100%;
  display: inline-flex;
  padding: 10px 12px;
  background-color: #f0f0f0;
  align-items: center;
  box-sizing: border-box;
  &__icon {
    margin-right: 12px;
  }
  &__clear-icon {
    margin-left: 10px;
    cursor: pointer;
  }
  svg {
    padding: 1px;
  }
  input {
    border: none;
    box-shadow: none;
    padding: 0px;
    box-sizing: border-box;
    width: 100%;
    background-color: #f0f0f0;
    font-size: 15px;
    &:focus {
      outline: none;
    }
  }
}
