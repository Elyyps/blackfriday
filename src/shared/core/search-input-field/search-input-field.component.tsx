import * as React from "react";
import styles from "./search-input-field-component.module.scss";
import SearchIcon from "@assets/icons/search.svg";
import CrossIcon from "@assets/icons/cross.svg";
import { IconComponent } from "../icon";
export interface ISearchInputFieldComponentProps {
  onChange?: (value: string) => void;
  placeholder?: string;
  value?: string;
}

const SearchInputFieldComponent = ({ value, placeholder, onChange }: ISearchInputFieldComponentProps) => (
  <div className={styles["search-input-field"]}>
    <div className={styles["search-input-field__icon"]}>
      <IconComponent size="14px" icon={SearchIcon} fillColor="#777" />
    </div>
    <input
      aria-label="Search input"
      value={value}
      placeholder={placeholder}
      onChange={event => {
        if (onChange) onChange(event.currentTarget.value);
      }}
    />
    <div
      onClick={() => {
        if (onChange) onChange("");
      }}
      role="button"
      style={{ opacity: value && value.length > 0 ? 1 : 0 }}
      className={styles["search-input-field__clear-icon"]}
    >
      <IconComponent size="14px" icon={CrossIcon} fillColor="#777" />
    </div>
  </div>
);

export { SearchInputFieldComponent };
