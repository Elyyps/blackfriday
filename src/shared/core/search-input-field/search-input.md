This component requires a value controlled by a variable (state) on the parent.
