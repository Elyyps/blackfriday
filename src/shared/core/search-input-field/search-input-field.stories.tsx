import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { SearchInputFieldComponent } from "./search-input-field.component";
const notes = require("./search-input.md");
storiesOf("SearchInputField", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => <SearchInputFieldComponent />, {
    notes
  });
