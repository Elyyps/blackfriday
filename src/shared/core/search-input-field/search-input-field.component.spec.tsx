import React from "react";
import { shallow } from "enzyme";
import { SearchInputFieldComponent } from "./search-input-field.component";

describe("[SearchInputField]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<SearchInputFieldComponent />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
