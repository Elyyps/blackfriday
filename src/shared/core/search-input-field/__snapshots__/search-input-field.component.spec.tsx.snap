// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[SearchInputField] should render component without crashing 1`] = `
<div
  className="search-input-field"
>
  <div
    className="search-input-field__icon"
  >
    <IconComponent
      fillColor="#777"
      icon="search.svg"
      size="14px"
    />
  </div>
  <input
    aria-label="Search input"
    onChange={[Function]}
  />
  <div
    className="search-input-field__clear-icon"
    onClick={[Function]}
    role="button"
    style={
      Object {
        "opacity": 0,
      }
    }
  >
    <IconComponent
      fillColor="#777"
      icon="cross.svg"
      size="14px"
    />
  </div>
</div>
`;
